### ISSUE TITLE
eglot error checking not working on .vue files

### ISSUE THREAD

**Issue by vikigenius** [hlissner/doom-emacs#4014]
I have installed the vls server and I am using the following snippet in my private config to set it up.

```
(define-derived-mode vk-vue-mode web-mode "vkVue"
  "A major mode derived from web-mode, for editing .vue files with LSP support.")
(add-to-list 'auto-mode-alist '("\\.vue\\'" . vk-vue-mode))
(add-hook 'vk-vue-mode-hook #'eglot-ensure)
(after! eglot
  (add-to-list 'eglot-server-programs
               '(vk-vue-mode . ("vls" "--stdio")))
  (add-to-list 'eglot-server-programs
               '(typescript-mode . ("typescript-language-server" "--stdio"))))
```

The server starts up automatically when I open .vue files, code completion etc. is okay.
However syntax/error checking is not working. The modeline status is stuck on the loading clock symbol, errors are not displayed.
flycheck-verify-setup shows that it is enabled and uses eglot. But it doesn't work for some reason.

**Comment by vikigenius** [comment]
I get the following warning when opening a .vue file

`Warning [flymake App.vue]: Disabling backend flymake-proc-legacy-flymake because (error Can’t find a suitable init function)
`
I am not sure if it's related though since flymake is not supposed to be used anyway?

**Comment by gagbo** [comment]
Probably fixed by #4471
